File: src/E022.hs
module E022 (e22) where

import Data.Char (ord)
import Data.List.Split (splitOn)
import Data.List (sort)

f :: [String] -> Int
f xs = sum [i * score name | (i, name) <- zip [1..] xs]

score ::  String -> Int
score name = sum [ord c - ord 'A'+1 | c <- name]

parse :: String -> [String]
parse = sort . splitOn "," . filter (/= '"')

e22 :: String -> Integer
e22 = fromIntegral . f . parse

File: src/E022.py
def load():
    with open('E022.txt', 'r') as f:
        return sorted(eval(f.read()))

def score(i, name):
    return i * sum([ord(c)-ord('A')+1 for c in name])

def f(data):
    return reduce(lambda a, b: a+score(b[0]+1, b[1]), enumerate(data), 0)

assert(score(938, 'COLIN') == 49714)
print f(load())
